test/cpp/jit/test_gpu.cpp

```diff
@@ -18148,6 +18148,135 @@ TEST(NVFuserTest, FusionRfactorContigIDs_CUDA) {
   testValidate(&fusion, outputs, aten_inputs, {ref}, __LINE__, __FILE__);
 }
 
+TEST(NVFuserTest, FusionMMASwizzlePrimitiveFloatAcc_CUDA) {
+  FusionExecutor fe;
+
+  std::string kernel = R"(
+  __global__ void mma32x32x8_mini_gemm_float(Tensor<__half, 2> Ag, Tensor<__half, 2> Bg, Tensor<float, 2> Cg0, Tensor<float, 2> Cg1){
+    // Accumulator
+    float C[32];
+    // initialize C:
+    for(int i=0;i<32;i++){
+        C[i] = 0;
+    }
+    // Allocate Smem
+    __shared__ uint4 As_mem[32*8 / 8];
+    __shared__ uint4 Bs_mem[8*32 / 8];
+    auto As = reinterpret_cast<__half*>(As_mem);
+    auto Bs = reinterpret_cast<__half*>(Bs_mem);
+    // Compute lane_id
+    int lane_id = threadIdx.x % 32;
+    
+    // Mini prolog: (single warp)
+        // global load:
+    __half Ag_buffer[8];
+    *reinterpret_cast<uint4*>(Ag_buffer) = *reinterpret_cast<uint4*>(&Ag[threadIdx.x*8]);
```

Comment:
Wouldn't it be simpler to use `uint4` rather than `__half`?

Reply:
Yes I can change that. Thanks for pointing this out 👍